[WebNN] Introduce operation types and inputs mapping between onnx and webnn

NingW101 opened the pull request:
### Description
Refactor the operation types and inputs mapping between onnx and webnn



### Motivation and Context
To simplify the maintenance of operation types and inputs, the map of onnx inputs name and webnn inputs name will be used to check inputs' rankrange

Honry commented on onnxruntime/core/providers/webnn/builders/helper.h:
```diff
@@ -209,146 +209,170 @@ const std::map<std::string_view, std::vector<std::string_view>> decomposed_op_ma
 // Retrieve the first input name of a WebNN op used for validating supported input data types.
 // WebNN ops have various first input names such as 'a', 'input', 'inputs', etc.
-// Special names other than 'input' are recorded in the webnn_op_first_input_name_map.
+// All WebNN op inputs are record in op_inputs_map
```
```suggestion
// All WebNN op inputs are recorded in op_inputs_map.
```

NingW101 commented on onnxruntime/core/providers/webnn/builders/op_map.h:
```diff
@@ -0,0 +1,189 @@
+
+#pragma once
+#include <map>
+#include "core/common/common.h"
```
@Honry not sure this include line of header file is right or not, which is to address the `onnx_to_webnn_data_type_map` & `supported_fallback_integer_data_types `

tianleiwu committed:
support more data types in tensor dumper (#24813)

### Description
support of dumping tensors of int8, uin t8, BFloat16, UInt4x2, and
Int4x2 data types in the tensor dumper.

### Motivation and Context
Help debugging of operators using these data types.

qti-jkilpatrick committed:
QNN EP can serialize graph to DLC (#24775)

### Description

This change adds support for serializing the QNN graph to the new Deep Learning Container (DLC) format. It is meant to supplement and perhaps eventually replace use of the QnnSaver backend, which emits C++ source
files when `qnn_saver_path` is set.

* Add support for serializing to .dlc via the QnnIr backend.
* Don't silently fallback to QnnCpu when QnnSaver was explicitly selected as the execution backend.
* Minor fixes.

### Motivation and Context

QNN model libraries, produced by compiling the C++ files that may be produced by QnnSaver have a number of drawbacks. Most importantly, they are not cross-platform and cannot be visualized via Netron or other tools. For these reasons, we anticipate that they may eventually be deprecated in favor of DLC files. These containers typically include a platform-agnostic representation of the graph QNN's internal representation.

---------

Co-authored-by: Jeff Kilpatrick <jkilpat@qti.qualcomm.com>

fs-eire committed:
[nodejs] fix nodejs cuda12 installation (#24814)

### Description
<!-- Describe your changes. -->

Fix Node.js linux/x64 cuda12 installation.

### Motivation and Context
<!-- - Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here. -->

toothache committed:
[TRT EP] Fix trt context memory sharing (#24784)

### Description
<!-- Describe your changes. -->

Fixed a TRT context memory sharing bug where the context memory was
assigned to a unique_ptr that was immediately destructed upon leaving
scope.


### Motivation and Context
<!-- - Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here. -->

The bug seems to be introduced by a refactor work: #15833 :


![image](https://github.com/user-attachments/assets/eec0e363-b6b1-4831-9ee4-a1b3ed45116c)

chenweng-quic committed:
[QNN EP] Add LSTM op builder for QNN EP (#24646)

### Description
Add LSTM support for QNN EP

### Motivation and Context
Add LSTM support for QNN EP

quic-tirupath committed:
[QNN EP] Fix 16x16 Conv translation (#24729)

- QNN's 16x16 Conv doesn't support asymmetric int16 weight
- Insert Convert Op to convert from asymmetric uint16 weight to symmetric int16 weight

### Description
- QNN' Conv op doesn't support asymmetric INT16 weights.
- 16x16 Conv operators in ONNX models fallback to CPU execution provider and reporting higher inference times.
- Insert a Convert Op to convert asymmetric uint16 weight to symmetric int16 weight to schedule 16x16 Conv's on QNN EP provider.



### Motivation and Context
- This fixes Graph execution failures for models contain 16x16 Conv op
on QNN Execution provider
- This also improves Inference times of model contain 16x16 Conv op

tianleiwu committed:
Remove unused tensor dumper functions (#24821)

### Description
Remove unused tensor dumper functions. 

Those functions are not needed any more since it is easy to make a
string with `::onnxruntime::MakeString` like in `DUMP_CPU_STRING`
macros.

### Motivation and Context
Follow up with
https://github.com/microsoft/onnxruntime/pull/24813#discussion_r2096803842.

Some functions were added, but not used any more. Remove them to avoid
maintenance cost.

kuanyul-quic committed:
[QNN EP] Fix inconsistent inputs for graph (#24751)

### Description

- Match the graph input correctly
- Add GetGraphInputNumber function

### Motivation and Context

- The number of graph inputs and the number of tensor wrappers may not match.
- For example, for ResizeNearestNeighbor op, Qnn only cares about the 1st input, so the rest of the inputs are not converted to tensor wrappers. However, these remaining inputs still appear in the graph inputs, resulting in a discrepancy in the input quantities.

ishwar-raut1 committed:
[NV TensorRt RTX EP] : Fix Domain check.  (#24816)

### Description
<!-- Describe your changes. -->
Small change to remove the MS Domain check on onnx model nodes


### Motivation and Context
<!-- - Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here. -->
The check returns unsupported for some nodes having an MS Domain. Trt
RTX supports some MS domain ops. if return unsupported these ops falls
back to CPU EP

@ankan-ban @chilo-ms @gedoensmax @jywu-msft

Co-authored-by: iraut <iraut@nvidia.com>

quic-ashwshan committed:
[QNN EP] MaxPool input rank-3 auto pad bug fix (#24827)

- Previously, padding for rank-3 MaxPool was only computed for auto_pad="NOTSET", using the final output shape.
- Identified a broader issue during auto_pad="VALID" implementation: padding must be derived from the recalculated output shape.
- Added unit tests to cover all use cases of auto_pad.
- Enabled the failing unit test in the cpu pool test

### Description
This PR fixes an issue in the padding calculation logic for rank-3 MaxPool operations when using auto_pad. The bug stemmed from using the final output shape (rank-3) to compute padding, rather than the correct intermediate shape (rank-4) that MaxPool actually operates on. The logic has been updated to use the reshaped rank-4 output for accurate padding
computation. Unit tests have been added to validate behavior across all auto_pad modes.

### Motivation and Context
While implementing support for auto_pad="VALID" in MaxPool, we discovered that the padding for MaxPool rank-3 was being calculated using the final output shape, which is rank-3. However, MaxPool internally operates on a reshaped rank-4 tensor (via pre- and post-processing reshapes). As a result, the padding logic was misaligned with the actual shape used during pooling, leading to test failures.

Honry reviewed (approved):
👍

HectorSVC committed:
Update Qnn default version to 2.34.0.250424 (#24750)

### Description
Update Qnn default version to 2.34.0.250424

fs-eire committed:
[build] disable vcpkg for Dawn temporarily (#24838)

### Description
<!-- Describe your changes. -->



### Motivation and Context
<!-- - Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here. -->

anujj committed:
Switch the TRT optimization profile if multi-profile is enable (#24805)

- Switch the multiple profile if multi profile is enabled. 
- Pass the profile index via OnRunStart option 
@ankan-ban @gaugarg-nv  @chilo-ms

tianleiwu committed:
Update MatMulBNits spec and Add Input Checks (#24828)

### Description

Major changes of spec:
* 2D scale shape: [N * n_blocks_per_col] => [N, n_blocks_per_col]
* 2D zero shape: [N * CeilDiv(n_blocks_per_col * bits, 8)] => [N,
CeilDiv(n_blocks_per_col * bits, 8)]
* For B, drop int32 type and only allow uint8.
* allow bfloat16 as input/output type.
* Mark input g_idx as deprecated (since it has no benefit on model size
and performance in inference).

Add a function CheckInputs to verify the input shape.

The reason of the shape change is to make scale and zero compatible with
other operators like DequantizeLinear and GatherBlockQuantized. That
will make it easy for graph fusion and model builder.

Note that ORT can still handle the legacy 1D format for scale and zero
points, and CUDA/CPU could still handle g_idx. However, they are
deprecated, and our tools shall generate 2D scale and zeros, and avoid
using g_idx going forward.

This change is backward compatible. Model from old spec can run in
latest ORT (CheckInputs handles 1D scale and zero points), and model
from latest spec can still run in older ORT (since older ORT does not
check dimension of scale and zero points)

### Motivation and Context

CUDA and CPU provider does not check inputs for MatMulNBits. It could
cause out of boundary access.

We are going to share the lm_head weights of MatMulNBits to
GatherBlockQuantized. 2D shape can be used in Gather directly, and we
can avoid Reshape nodes.

Our latest models published for foundry use 2D scale and zero points. So
I update the spec to reflect that.

kevinch-nv committed:
[TRT EP] Update build and API usage for TensorRT 10.11 (#24832)

### Description

Resolves the following issues starting in TensorRT 10.11:
- Version macros changed in `NvInferVersion.h`, update build to look for
new macros
- Updated deprecated APIs (setShapeValues -> setShapeValuesV2() to
support INT64 shape values)



### Motivation and Context

- Resolves building TensorRT EP from source with latest 10.11 release.

Signed-off-by: Kevin Chen <kevinch@nvidia.com>

gedoensmax committed:
[NvTensorRT RTX] Add Bfloat16 (#24743)

### Description

TRT supports Bfloat 16 and ORT does as well. 
In addition the `setup.py` was missing a copy for NVTRT EP and TRT EP
can only be built against the packaged parser with TRT RTX.

quic-tirupath committed:
[QNN EP] Add ScatterND reduction attribute (#24844)

### Description
 - Add support for ScatterND reduction attribute
 - Gracefully handle the unsupported reduction values
 - Add unit tests to validate Reduction attribute support

### Motivation and Context
<!-- - Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here. -->

NingW101 committed:
[WebNN] Refactor op mappings and add input name mapping between ONNX and WebNN (#24830)

### Description
Add `map_info.h` to centralize the operation types and inputs mapping
between onnx and webnn.

### Motivation and Context
To simplify the maintenance of operation types and inputs. The mapping
of onnx input names and webnn input names will be used in the future to
check the `rankRange`.



@honry, @fdwr, @guschmue, PTAL, thanks!

---------

Co-authored-by: Wanming Lin <wanming.lin@intel.com>

xiaofeihan1 committed:
[Mac] Fix --use_xcode build with Nodejs binding (#24868)

### Description
<!-- Describe your changes. -->

Currently, the XCode build with nodejs binding(`--use_xcode`) always
fails on Mac.
```
./build.sh --config Debug --use_xcode --use_webgpu --build_shared_lib --build_nodejs --parallel --compile_no_warning_as_error --skip_submodule_sync --cmake_extra_defines CMAKE_OSX_ARCHITECTURES=arm64 --skip_tests
```
The root cause is that the dylib locates on `/Debug/Debug` not `/Debug`
with using XCode generator. For other generator(e.g. make, ninja), the
dylib locates on `/Debug` as expected.

Mac pipeline can pass because they didn't use XCode generator.

<img width="913" alt="image"
src="https://github.com/user-attachments/assets/e1203fdb-d88a-4c06-abad-b641d502237c"
/>

dependabot[bot] committed:
Bump setuptools from 69.0.3 to 78.1.1 in /tools/ci_build/github/linux/docker/scripts (#24810)

Bumps [setuptools](https://github.com/pypa/setuptools) from 69.0.3 to
78.1.1.
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/pypa/setuptools/blob/main/NEWS.rst">setuptools's
changelog</a>.</em></p>
<blockquote>
<h1>v78.1.1</h1>
<h2>Bugfixes</h2>
<ul>
<li>More fully sanitized the filename in PackageIndex._download. (<a
href="https://redirect.github.com/pypa/setuptools/issues/4946">#4946</a>)</li>
</ul>
<h1>v78.1.0</h1>
<h2>Features</h2>
<ul>
<li>Restore access to _get_vc_env with a warning. (<a
href="https://redirect.github.com/pypa/setuptools/issues/4874">#4874</a>)</li>
</ul>
<h1>v78.0.2</h1>
<h2>Bugfixes</h2>
<ul>
<li>Postponed removals of deprecated dash-separated and uppercase fields
in <code>setup.cfg</code>.
All packages with deprecated configurations are advised to move before
2026. (<a
href="https://redirect.github.com/pypa/setuptools/issues/4911">#4911</a>)</li>
</ul>
<h1>v78.0.1</h1>
<h2>Misc</h2>
<ul>
<li><a
href="https://redirect.github.com/pypa/setuptools/issues/4909">#4909</a></li>
</ul>
<h1>v78.0.0</h1>
<h2>Bugfixes</h2>
<ul>
<li>Reverted distutils changes that broke the monkey patching of command
classes. (<a
href="https://redirect.github.com/pypa/setuptools/issues/4902">#4902</a>)</li>
</ul>
<h2>Deprecations and Removals</h2>
<ul>
<li>Setuptools no longer accepts options containing uppercase or dash
characters in <code>setup.cfg</code>.</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/pypa/setuptools/commit/8e4868a036b7fae3208d16cb4e5fe6d63c3752df"><code>8e4868a</code></a>
Bump version: 78.1.0 → 78.1.1</li>
<li><a
href="https://github.com/pypa/setuptools/commit/100e9a61ad24d5a147ada57357425a8d40626d09"><code>100e9a6</code></a>
Merge pull request <a
href="https://redirect.github.com/pypa/setuptools/issues/4951">#4951</a></li>
<li><a
href="https://github.com/pypa/setuptools/commit/8faf1d7e0ca309983252e4f21837b73ee12e960f"><code>8faf1d7</code></a>
Add news fragment.</li>
<li><a
href="https://github.com/pypa/setuptools/commit/2ca4a9fe4758fcd39d771d3d3a5b4840aacebdf7"><code>2ca4a9f</code></a>
Rely on re.sub to perform the decision in one expression.</li>
<li><a
href="https://github.com/pypa/setuptools/commit/e409e8002932f2b86aae7b1abc8f8c2ebf96df2c"><code>e409e80</code></a>
Extract _sanitize method for sanitizing the filename.</li>
<li><a
href="https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b"><code>250a6d1</code></a>
Add a check to ensure the name resolves relative to the tmpdir.</li>
<li><a
href="https://github.com/pypa/setuptools/commit/d8390feaa99091d1ba9626bec0e4ba7072fc507a"><code>d8390fe</code></a>
Extract _resolve_download_filename with test.</li>
<li><a
href="https://github.com/pypa/setuptools/commit/4e1e89392de5cb405e7844cdc8b20fc2755dbaba"><code>4e1e893</code></a>
Merge <a
href="https://github.com/jaraco/skeleton">https://github.com/jaraco/skeleton</a></li>
<li><a
href="https://github.com/pypa/setuptools/commit/3a3144f0d2887fa37c06550f42a101e9eebd953a"><code>3a3144f</code></a>
Fix typo: <code>pyproject.license</code> -&gt;
<code>project.license</code> (<a
href="https://redirect.github.com/pypa/setuptools/issues/4931">#4931</a>)</li>
<li><a
href="https://github.com/pypa/setuptools/commit/d751068fd2627d6d8f1729e39cbcd8119049998f"><code>d751068</code></a>
Fix typo: pyproject.license -&gt; project.license</li>
<li>Additional commits viewable in <a
href="https://github.com/pypa/setuptools/compare/v69.0.3...v78.1.1">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=setuptools&package-manager=pip&previous-version=69.0.3&new-version=78.1.1)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)
You can disable automated security fix PRs for this repo from the
[Security Alerts
page](https://github.com/microsoft/onnxruntime/network/alerts).

</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
